sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -52,6 +52,21 @@
 from .signatures import *
 
 
+@arange.override()
+def arange_replicated(
+    *args,
+    devices: Sequence[int] | None = None,
+    **kwargs,
+):
+    if devices is None:
+        return NotImplemented
+
+    # Do not use `tranfer_to_logical_device` here.
+    # Adding the transfer op would prevent the arange result from being fused.
+    shards = [arange(*args, **kwargs) for _ in devices]
```

Comment:
Would we support at some point here calling a custom arange function that constructs on the required device explicitly. Something that exports to an MLIR op that has the device affinity set.

Reply:
Being able to tag these 0 input ops and explicitly placing them is a good idea and would help not rely as strongly on IREE's affinity analysis. I don't know if that's possible at present, but if it is, it'll slot in easily here.